Sources/PackageGraph/RepositoryPackageContainerProvider.swift

```diff
@@ -49,6 +47,19 @@ public class RepositoryPackageContainerProvider: PackageContainerProvider {
         // Create the container wrapper.
         return RepositoryPackageContainer(identifier: identifier, repository: repository, manifestLoader: manifestLoader)
     }
+
+    public func getContainer(for identifier: Container.Identifier, completion: @escaping (Result<Container, AnyError>) -> Void) {
+        repositoryManager.lookup(repository: identifier) { result in
```

Comment:
This seems fine, but I wonder if we eventually are going to want the resolver to maintain its own map of all the containers it has looked up, and their status). So prefetch would initial an entry in this map, and then when we *have* to have it we would wait on it becoming available...